### PR TITLE
Add 6-hrly max winds at 10m to SFS

### LINKED ISSUE #986: Add temporal max wind at 10-meter as SFS product
CPC requests temporal max wind at 10-meter as SFS product.

### PR DESCRIPTION
**Pull request by KarinaAsmar-NOAA**
This PR addresses [Issue #986](https://github.com/NOAA-EMC/UPP/issues/986) for adding 6-hourly max winds (u,v, and speed) at 10m, as requested by CPC. GRIB2 file with outputs is in WCOSS2: /lfs/h2/emc/vpppg/noscrub/karina.asmar/max_wind/post_sfs_2022090100/GFSPRS.GrbF24.

### CODE REVIEW

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 1]
```diff
@@ -2004,6 +2005,22 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
...
+      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
+      spval,VarName,wspd10umax)
+     if(debugprint)print*,'sample ',VarName,' = ',u10max(isa,jsa)
+! max 6-hourly v comp of 10m agl wind
+      VarName='v10m_max'
+      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
+      spval,VarName,wspd10vmax)
+     if(debugprint)print*,'sample ',VarName,' = ',v10max(isa,jsa)
+! max 6-hourly 10m agl wind speed
+      VarName='wind10m_max'
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f]: @KarinaAsmar-NOAA Reading 'wind10m_max" can be implemented in line 2003. You should make changes for reading 'spf10max' for MODELNAME 'FV3R' and 'wind10m_max' for MODELNAME 'GFS'.

Comment (KarinaAsmar-NOAA) [reply sorc/ncep_post.fd/INITPOST_NETCDF.f]: @WenMeng-NOAA The change has been pushed to the branch. Results are in WCOSS2: /lfs/h2/emc/vpppg/noscrub/karina.asmar/max_wind/ -> post_sfs_2022090100 has grib2 results for SFS/GFS variable name, post_fv3r_2023062800 has results with FV3R variable name: 758:711064838:d=2023062800:WIND:10 m above ground:9-10 hour max fcst

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 2]
```diff
@@ -56,6 +56,7 @@
 !> 2024-05-09 | Eric James    | Enable reading of clear-sky downwelling shortwave irradiance
 !> 2024-05-10 | Karina Asmar  | Read omega from model output and calculate HGT for hydrostatic runs
 !> 2024-06-25 | Wen Meng      | Add capability to read fhzero as either an integer or float
+!> 2024-08-26 | Karina Asmar  | Add 6-hourly u/v, speed max wind components at 10m agl
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f]: @KarinaAsmar-NOAA We would like these temporal max 10m wind variables used for any UFS-based applications. Please remove '6-houly' from notes.

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 3]
```diff
@@ -1998,12 +1999,27 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
...
+      if (modelname=='FV3R') then
+        VarName='spd10max'  ! hourly max wind speed at 10m
+      else if (modelname=='GFS') then
+        VarName='wind10m_max' ! 6-hourly max wind speed at 10m
+      endif
       call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
       spval,VarName,wspd10max)
      if(debugprint)print*,'sample ',VarName,' = ',wspd10max(isa,jsa)
 
+! max 6-hourly u comp of 10m agl wind
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f]: @KarinaAsmar-NOAA Remove '6-hourly'

File: sorc/ncep_post.fd/INITPOST_NETCDF.f [hunk 4]
```diff
@@ -1998,12 +1999,27 @@ SUBROUTINE INITPOST_NETCDF(ncid2d,ncid3d)
...
+      VarName='u10m_max'
+      call read_netcdf_2d_para(ncid2d,ista,ista_2l,iend,iend_2u,jsta,jsta_2l,jend,jend_2u, &
+      spval,VarName,wspd10umax)
+     if(debugprint)print*,'sample ',VarName,' = ',u10max(isa,jsa)
+! max 6-hourly v comp of 10m agl wind
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_NETCDF.f]: @KarinaAsmar-NOAA Remove '6-hourly'

Comment (KarinaAsmar-NOAA) [reply sorc/ncep_post.fd/INITPOST_NETCDF.f]: @WenMeng-NOAA 6-hourly has been replace with 'temporal' now.

File: sorc/ncep_post.fd/SURFCE.f [hunk 5]
```diff
@@ -2621,12 +2641,8 @@ SUBROUTINE SURFCE
          if(grib=='grib2') then
            cfld=cfld+1
            fld_info(cfld)%ifld=IAVBLFLD(IGET(422))
-           if (ifhr==0) then
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f:2624]: @KarinaAsmar-NOAA Please keep the original setting for modelname 'FV3R' which is hard-wired for 1-hourly max, and new changes for flexible bucket.

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/SURFCE.f:2624]: @KarinaAsmar-NOAA The original setting should also be applied to modelname 'RAPR'.

Comment (KarinaAsmar-NOAA) [reply sorc/ncep_post.fd/SURFCE.f:2624]: It should be done now.

File: sorc/ncep_post.fd/SURFCE.f [hunk 6]
```diff
@@ -2609,9 +2610,33 @@ SUBROUTINE SURFCE
 !
 ! SRD
 !
-!        ANEMOMETER LEVEL (10 M) MAX WIND SPEED.
+!        ANEMOMETER LEVEL (10 M) MAX WINDS.
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f]: @KarinaAsmar-NOAA Please keep the original note "MAX WIND SPEED"

File: sorc/ncep_post.fd/SURFCE.f [hunk 7]
```diff
@@ -2609,9 +2610,33 @@ SUBROUTINE SURFCE
...
 !
-      IF (IGET(422)>0) THEN
+      IF ((IGET(422)>0) .OR. (IGET(783)>0) .OR. (IGET(784)>0)) THEN
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f]: @KarinaAsmar-NOAA Two setting options should be applied for max wind speed only. You may remove "IGET(783)>0" and "IGET(784)>0".

File: sorc/ncep_post.fd/SURFCE.f [hunk 8]
```diff
@@ -2650,12 +2697,16 @@ SUBROUTINE SURFCE
          if(grib=='grib2') then 
            cfld=cfld+1
            fld_info(cfld)%ifld=IAVBLFLD(IGET(783))
-           if (ifhr==0) then 
+           fld_info(cfld)%ntrange=1
+           IF (MODELNAME == 'FV3R' .OR. MODELNAME == 'RAPR') THEN
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f]: @KarinaAsmar-NOAA It turns out that RRFS doesn't use this block to output their products. You may remove the setting "MODELNAME=='FV3R' ".

File: sorc/ncep_post.fd/SURFCE.f [hunk 9]
```diff
@@ -2679,12 +2751,16 @@ SUBROUTINE SURFCE
          if(grib=='grib2') then
            cfld=cfld+1
            fld_info(cfld)%ifld=IAVBLFLD(IGET(784))
-           if (ifhr==0) then
+           fld_info(cfld)%ntrange=1
+           IF (MODELNAME == 'FV3R' .OR. MODELNAME == 'RAPR') THEN
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f]: @KarinaAsmar-NOAA It turns out that RRFS doesn't use this block to output their products. You may remove the setting "MODELNAME=='FV3R' ".

Comment (KarinaAsmar-NOAA) [reply sorc/ncep_post.fd/SURFCE.f]: @WenMeng-NOAA Remove MODELNAME=='FV3R' for all 3 products (ids 422, 783, 784)?

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/SURFCE.f]: @KarinaAsmar-NOAA Only 783 and 784.

**Comment by WenMeng-NOAA** [comment]
@KarinaAsmar-NOAA Your PR looks good to me now. We'll proceed with final testing.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You may start the UPP RTs on R&D machines. There should be no baseline updates with this PR.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2 without baseline changes.

**Review by FernandoAndrade-NOAA — APPROVED** [review]
No changes across Hera, Hercules, and Orion RTs.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**